test/e2e/service_externalip_test.go

```diff
@@ -779,6 +779,26 @@ func testExternalIPAccess(t *testing.T, data *TestData) {
 					externalIP, host, err := waitExternalIPConfigured(service)
 					require.NoError(t, err)
 
+					// A route to the client IP is required for DSR mode.
+					nodeInfo := getNodeByName(host)
+					workerNodeIP := clusterInfo.nodes[nodeInfo.idx].ip()
+					addRouteToClientIPCmd := []string{"ip", "route", "replace", tt.clientIP, "via", workerNodeIP}
+					delRouteToClientIPCmd := []string{"ip", "route", "del", tt.clientIP, "via", workerNodeIP}
+
+					for idx := range nodes {
+						node := nodes[idx]
+						if node == host {
+							continue
+						}
+						stdout, stderr, err := data.RunCommandFromAntreaPodOnNode(node, addRouteToClientIPCmd)
+						require.NoError(t, err, "Failed to add route to client IP on Node %s, stdout: %s, stderr: %s: cmd: %q", node, stdout, stderr, addRouteToClientIPCmd)
```

Comment:
The command failed at hybrid mode: 
```
Failed to add route to client IP on Node kind-control-plane, stdout: , stderr: Error: Nexthop has invalid gateway.
        	            	: cmd: ["ip" "route" "replace" "169.254.170.1" "via" "20.20.20.3"]
```
You may need to specify the interface explicitly to make the cmd work.

Reply:
It fails because we cannot add the route in hybrid mode. I added a check to ensure this route will be added only in DSR mode as DSR itself can only work with encap mode.